src/components/MnemonicPhraseInput.tsx

```diff
@@ -18,17 +42,65 @@ export default function MnemonicPhraseInput({
 }: MnemonicPhraseInputProps) {
   const [activeIndex, setActiveIndex] = useState(0)
   const inputRefs = useRef<HTMLInputElement[]>([])
+  const [showDropdown, setShowDropdown] = useState<number | null>(null)
+  const [filteredWords, setFilteredWords] = useState<string[]>([])
+  const dropdownRef = useRef<HTMLDivElement>(null)
+
+  const focusNextInput = useCallback((index: number) => {
+    inputRefs.current[index]?.focus()
+  }, [])
 
   useEffect(() => {
     if (activeIndex < mnemonicPhrase.length && isValid && isValid(activeIndex)) {
       const nextIndex = activeIndex + 1
       setActiveIndex(nextIndex)
+      setShowDropdown(null)
+      focusNextInput(nextIndex)
+    }
+  }, [mnemonicPhrase, activeIndex, isValid, focusNextInput])
+
+  const updateFilteredWords = useCallback((value: string, index: number) => {
+    const matched = value ? MNEMONIC_WORDS.filter((word) => word.startsWith(value)) : []
+    setShowDropdown(matched.length > 0 ? index : null)
+    setFilteredWords(matched)
+  }, [])
 
-      if (inputRefs.current[nextIndex]) {
-        inputRefs.current[nextIndex].focus()
+  const handleInputChange = useCallback(
+    (value: string, index: number, selectWordFromDropdown = false) => {
+      const newPhrase = mnemonicPhrase.map((word, i) => (i === index ? value : word))
+      onChange(newPhrase)
+      if (selectWordFromDropdown) {
+        setShowDropdown(null)
+        if (!isValid) {
+          focusNextInput(index + 1)
+        }
+      } else {
+        updateFilteredWords(value, index)
       }
-    }
-  }, [mnemonicPhrase, activeIndex, isValid])
+    },
+    [mnemonicPhrase, onChange, isValid, focusNextInput, updateFilteredWords],
+  )
+
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent, value: string, wordIndex: number) => {
+      if (e.key === 'ArrowDown') {
+        e.preventDefault()
+        if (filteredWords.length > 0 && dropdownRef.current) {
+          const firstItem = dropdownRef.current.querySelector('.dropdown-item')
+          if (firstItem) {
+            ;(firstItem as HTMLElement).focus()
+          }
+        }
+      } else if (e.key === 'Enter') {
+        const matched = MNEMONIC_WORDS.filter((word) => word.startsWith(value))
+        if (matched.length === 1) {
+          handleInputChange(matched[0], wordIndex, true)
+          e.preventDefault()
```

Comment:
```suggestion
          e.preventDefault()
          handleInputChange(matched[0], wordIndex, true)
```

Reply:
fixed in https://github.com/joinmarket-webui/jam/pull/835/commits/dde55148f5c9ab7de15b33be5e2977f9f9adce06